### DISCUSSION TITLE
How do I compile an eide project with powershell？How to generate builder.params use powershell cmd?

### DISCUSSION THREAD

**Discussion by gaolonglong11** [github0null/eide#285] [Q&A]
if use edie build one time ; builder.params generate;

**Comment by github0null** [comment]
use this command:

```
unify_builder -p xxxxx\xxxx\YOUR\PATH\builder.params
```

notice:

before execute this command, please make sure that we can find `C:\Users\<user_name>\.eide\bin\builder\bin\unify_builder.exe` in your system `PATH` variables

**Reply by gaolonglong11** [reply]
**How to generate builder.params by command**

**Reply by gaolonglong11** [reply]
I want to automate compilation with EIDE+GCC; Whether EIDE can automatically generate makefile of project?

**Reply by github0null** [reply]
暂时不支持用命令行生成 builder.params

但 builder.params 是一个 json 文件，是可读写的，里面描述了这个项目在编译时的所有输入

你可以先用插件生成一个 builder.params 模板，然后用 shell 脚本去修改该文件，最后传递给 unify_builder.exe

**Reply by github0null** [reply]
makefile 的生成还在计划中，预计生成的出的 makefile 也是类似与 eclipse 风格的，每个文件一个 rule 那样的
生成出的 makefile 可读性较差，是很难去手动修改的

**Reply by vscodeenv** [reply]
可读性我觉得问题不大，主要是能够准确高效的makefile就可以，希望能够通过命令实现自动化编译

**Reply by vscodeenv** [reply]
我的想法是这个环境能够支持代码在linux环境下面进行自动编译，要自动编译就得自己用cmake或者make命令；在windows下面是可以直接用默认的图形界面进行编译，调试和下载

**Reply by github0null** [reply]
在现在的这个工具的对应关系：

 unify_builder.exe 就等同于 make.exe
 
 builder.params 就等同于 Makefile 文件
 
其实也没太大差别，unify_builder 也是有 linux 版本的

**Reply by vscodeenv** [reply]
理解，这个unify_builder在linux里面安装以后还是需要先生成builder.params传给linux里面才可以进行通过unify_builder编译吧？
unify_builder这个linux版本你可以给个链接或者使用说明吗，多谢！

**Reply by github0null** [reply]
1. 你需要在 linux 上的 vscode 上安装该插件和 dotnet 6 运行时环境（就跟 windows 上的一样）

2. 然后 unify_builder 会被安装到用户家目录下，`~/.eide/bin/builder/bin`，你需要将这个路径加到 PATH 中

3. builder.params 需要先用插件的 build 按钮生成一次，然后这个文件你可以自己保存到其他地方，方便修改

   builder.params 的内容，目前没有介绍文档，json字段的名字能看出大概含义

5. 使用 `unify_builder -h` 查看帮助，如果只是编译的话：`unify_builder -p <builder.params 的路径>` 就行了

**Reply by vscodeenv** [reply]
您好！
可以请教一下，builder.params的生成必须是通过点击按钮来产生吗？能不能通过命令生成这个builder.params
